lib/gpu/lib/src/render_pass.dart

```diff
@@ -21,6 +21,32 @@ base class ColorAttachment {
 
   Texture texture;
   Texture? resolveTexture;
+
+  void _assertValid() {
```

Comment:
I think this would be easier to read if you wrote it as a regular error checking function, throwing exceptions, then only called the function in an assert.

This is a regular pattern used in the framework, for example:  https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/animation/animation_controller.dart#L551-L564

Reply:
Thanks. I updated this. Let me know if you think I can improve the pattern more.